File: common/base_components/src/main/java/ru/tensor/sbis/base_components/adapter/ItemClickListener.java
package ru.tensor.sbis.base_components.adapter;

/**
 * Интерфейс обработчика клика по элементу списка.
 *
 * @author am.boldinov
 */
public interface ItemClickListener {

    /**
     * Обработка нажатия указанного элемента списка.
     * @param position - позиция элемента в списке
     */
    void onItemClick(int position);
}

File: design/selection/src/main/java/ru/tensor/sbis/design/selection/ui/contract/listeners/ItemClickListener.kt
package ru.tensor.sbis.design.selection.ui.contract.listeners

import androidx.fragment.app.FragmentActivity
import ru.tensor.sbis.design.selection.ui.model.SelectorItemModel
import java.io.Serializable

/**
 * Пользовательский обработчик нажатий на области ячейки
 *
 * @author ma.kolpakov
 */
interface ItemClickListener<DATA : SelectorItemModel, ACTIVITY : FragmentActivity> : Serializable {

    /**
     * Короткое или долгое нажатие в зависимости от места применения
     *
     * @see SelectorItemListeners
     */
    fun onClicked(activity: ACTIVITY, item: DATA)
}